src/Columns/ColumnAggregateFunction.h

```diff
@@ -82,6 +84,10 @@ class ColumnAggregateFunction final : public COWHelper<IColumn, ColumnAggregateF
     /// Name of the type to distinguish different aggregation states.
     String type_string;
 
+    /// MergedData records, used to avoid duplicated data copy.
+    ///key: src pointer, val:  pos in current column.
+    std::unordered_map<ConstAggregateDataPtr, size_t> copiedDataInfo;
```

Comment:
It should be better to use `HashMap` from `Common/HashTable/HashMap.h`, because it's more performant.

Reply:
2021-02-03 11:50:15 /ClickHouse/src/Common/HashTable/HashTable.h:72:37: error: zero as null pointer constant [-Werror,-Wzero-as-null-pointer-constant]
2021-02-03 11:50:15 bool check(const T x) { return x == 0; }
2021-02-03 11:50:15                                     ^
2021-02-03 11:50:15                                     nullptr
2021-02-03 11:50:15 /ClickHouse/src/Common/HashTable/HashMap.h:72:87: note: in instantiation of function template specialization 'ZeroTraits::check<const char *>' requested here
2021-02-03 11:50:15     static bool isZero(const Key & key, const State & /*state*/) { return ZeroTraits::check(key); }
2021-02-03 11:50:15                                                                                       ^
2021-02-03 11:50:15 /ClickHouse/src/Common/HashTable/HashMap.h:71:53: note: in instantiation of member function 'HashMapCell<const char *, unsigned long, DefaultHash<const char *, void>, HashTableNoState>::isZero' requested here
2021-02-03 11:50:15     bool isZero(const State & state) const { return isZero(value.first, state); }
2021-02-03 11:50:15                                                     ^
2021-02-03 11:50:15 /ClickHouse/src/Common/HashTable/HashTable.h:741:38: note: in instantiation of member function 'HashMapCell<const char *, unsigned long, DefaultHash<const char *, void>, HashTableNoState>::isZero' requested here
2021-02-03 11:50:15         while (ptr < buf_end && ptr->isZero(*this))
2021-02-03 11:50:15                                      ^
2021-02-03 11:50:15 /ClickHouse/src/Common/HashTable/HashTable.h:525:32: note: in instantiation of member function 'HashTable<const char *, HashMapCell<const char *, unsigned long, DefaultHash<const char *, void>, HashTableNoState>, DefaultHash<const char *, void>, HashTableGrower<3>, AllocatorWithStackMemory<Allocator<true, true>, 128, 1> >::begin' requested here
2021-02-03 11:50:15             for (iterator it = begin(), it_end = end(); it != it_end; ++it)

Reply:
hash map compile failed.